This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108164` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>solanum</li>
  </ul>
</details>

```
----------------------------------------------------------------------
make[3]: Leaving directory '/build/source/extensions'
make[2]: Leaving directory '/build/source/extensions'
Making install in help
make[2]: Entering directory '/build/source/help'
building index files
/nix/store/4l7wsi6h6283194r6fqy1731qxlagq62-bash-4.4-p23/bin/bash: line 4: column: command not found
/nix/store/4l7wsi6h6283194r6fqy1731qxlagq62-bash-4.4-p23/bin/bash: line 4: column: command not found
tr: write error: Broken pipe
ircd: setting up help directory structure
make[2]: Leaving directory '/build/source/help'
Making install in doc
make[2]: Entering directory '/build/source/doc'
mkdir -p /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/etc
ircd: installing example config files (ircd.conf.example reference.conf)
ircd: installing motd file (ircd.motd)
make[2]: Leaving directory '/build/source/doc'
make[2]: Entering directory '/build/source'
make[3]: Entering directory '/build/source'
make  install-exec-hook
make[4]: Entering directory '/build/source'
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/*.la
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/solanum/modules/*.la
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/solanum/modules/autoload/*.la
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/solanum/modules/extensions/*.la
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/*.dll.a
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/solanum/modules/*.dll.a
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/solanum/modules/autoload/*.dll.a
rm -f /nix/store/ypmpcc53xxld1izhkxfichj6bcc2ppmw-solanum-unstable-2020-12-14/lib/solanum/modules/extensions/*.dll.a
make[4]: Leaving directory '/build/source'
make  install-data-hook
make[4]: Entering directory '/build/source'
test -d /var/lib/solanum/log/solanum || mkdir -p /var/lib/solanum/log/solanum
mkdir: cannot create directory '/var': Permission denied
make[4]: *** [Makefile:921: install-data-hook] Error 1
make[4]: Leaving directory '/build/source'
make[3]: *** [Makefile:834: install-data-am] Error 2
make[3]: Leaving directory '/build/source'
make[2]: *** [Makefile:785: install-am] Error 2
make[2]: Leaving directory '/build/source'
make[1]: *** [Makefile:475: install-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:778: install] Error 2
```

--- a/pkgs/servers/irc/solanum/default.nix
+++ b/pkgs/servers/irc/solanum/default.nix
@@ -1,0 +1,60 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, bison
+, flex
+, openssl
+, sqlite
+, lksctp-tools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "solanum";
+  version = "unstable-2020-12-14";
+
+  src = fetchFromGitHub {
+    owner = "solanum-ircd";
+    repo = pname;
+    rev = "551e5a146eab4948ce4a57d87a7f671f2d7cc02d";
+    sha256 = "14cd2cb04w6nwck7q49jw5zvifkzhkmblwhjfskc2nxcdb5x3l96";
+  };
+
+  patches = [
+    ./dont-create-logdir.patch
+  ];
+
+  configureFlags = [
+    "--enable-epoll"
+    "--enable-ipv6"
+    "--enable-openssl=${openssl.dev}"
+    "--with-program-prefix=solanum-"
+    "--localstatedir=/var/lib/solanum"
+    "--with-rundir=/run/solanum"
+    "--with-logdir=/var/log/solanum"
+  ] ++ stdenv.lib.optionals (stdenv.isLinux) [
+    "--enable-sctp=${lksctp-tools.out}/lib"
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    bison
+    flex
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    sqlite
+  ];
+
+  doCheck = !stdenv.isDarwin;
+
+  meta = with stdenv.lib; {
+    description = "An IRCd for unified networks";
+    homepage = "https://github.com/solanum-ircd/solanum";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ hexa ];
+    platforms = platforms.unix;
+  };
+}

--- a/pkgs/servers/irc/solanum/dont-create-logdir.patch
+++ b/pkgs/servers/irc/solanum/dont-create-logdir.patch
@@ -1,0 +1,14 @@
+diff --git a/Makefile.am b/Makefile.am
+index 19e7b396..21093521 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -35,9 +35,6 @@ include/serno.h:
+ 		echo '#define DATECODE 0UL' >>include/serno.h; \
+ 	fi
+ 
+-install-data-hook:
+-	test -d ${DESTDIR}${logdir} || mkdir -p ${DESTDIR}${logdir}
+-
+ install-exec-hook:
+ 	rm -f ${DESTDIR}${libdir}/*.la
+ 	rm -f ${DESTDIR}${moduledir}/*.la

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7528,6 +7528,8 @@ in
 
   solaar = callPackage ../applications/misc/solaar {};
 
+  solanum = callPackage ../servers/irc/solanum {};
+
   sourceHighlight = callPackage ../tools/text/source-highlight { };
 
   spacebar = callPackage ../os-specific/darwin/spacebar {

